### PR TITLE
feat: `get_directory_file_info` for exploring a directory of files

### PR DESCRIPTION
**Pull request by MthwRobinson**
### Summary

Adds a `get_directory_file_info` helper function that summarizes the files in a directory. The function will recursively walk sub-directories. Also adds support for detecting additional filetypes that I encountered while testing the exploration function on a real-world set of documents.

### Testing

```python
from unstructured.file_utils.exploration import get_directory_file_info

file_info = get_directory_file_info("example-docs")
file_info.filetype.value_counts()
```

The result should look like:

```python
FileType.EML     4
FileType.TXT     3
FileType.HTML    2
FileType.XML     2
FileType.PDF     2
FileType.DOCX    1
FileType.PPTX    1
FileType.XLSX    1
FileType.JPG     1
Name: filetype, dtype: int64
```

### CODE REVIEW

File: unstructured/file_utils/filetype.py [hunk 1]
```diff
@@ -27,16 +37,61 @@
...
-    DOCX = 2
-    JPG = 3
-    TXT = 4
-    EML = 5
-    XML = 6
-    HTML = 7
-    XLSX = 8
+    UNK = 0
+
+    # MS Word Types
```
Comment (qued) [inline unstructured/file_utils/filetype.py:50]:
```suggestion
    # MS Office Types
```

File: docs/source/examples.rst [hunk 2]
```diff
@@ -270,3 +270,39 @@ You can also pass in a file-like object with:
...
 To extract metadata from ``.docx`` or ``.xlsx``, use ``get_docx_metadata`` and
 ``get_xlsx_metadata``. The interfaces are the same as ``get_jpg_metadata``.
+
+
+###########################
+Exploring Source Documents
+###########################
+
+The ``unstructured`` library includes tools for helping you explore source documents.
+To get a summary of the size and type of documents in a directory, you can
```
Comment (qued) [inline docs/source/examples.rst:280]: What's meant by "size" here? I don't see file sizes in the summary...

Comment (MthwRobinson) [reply docs/source/examples.rst:280]: If you run `file_info.head()` you'll see it. That's the size of the file in bytes. I'll update the docs to clarify

**Review by qued — APPROVED** [review]
LGTM!

**Pull request merged**